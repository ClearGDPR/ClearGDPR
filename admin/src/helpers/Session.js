import cookie from 'cookie';

export class Session {
  set(authResult) {
    const maxAge = 1 * 24 * 60 * 60; // 1 day long;
    document.cookie = cookie.serialize('auth', authResult, { maxAge });
    document.cookie = cookie.serialize('expires_at', authResult, { maxAge });
  }

  destroy() {
    document.cookie = cookie.serialize('auth', '', { maxAge: -1 });
    document.cookie = cookie.serialize('expires_at', '', { maxAge: -1 });
  }

  isLoggedIn() {
    let expiresAt = cookie.parse(document.cookie).expires_at || '';
    return new Date().getTime() < expiresAt;
  }
}

export default new Session();
