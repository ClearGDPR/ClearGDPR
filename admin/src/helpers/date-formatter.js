import { format } from 'date-fns';

export default date => {
  return format(new Date(date), 'DD/MM/YYYY h:mma');
};
