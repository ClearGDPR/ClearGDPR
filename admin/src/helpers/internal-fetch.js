import merge from 'lodash/object/merge';
import session from '../../helpers/Session';

const internalFetch = (url, config = {}) => {
  const internalConfig = merge(
    {
      method: 'GET',
      headers: {
        Authorization: `Bearer ${session.getToken()}`,
        'Content-Type': 'application/json'
      }
    },
    config
  );
  return fetch(url, internalConfig).then(async res => {
    if (res.status === 400 && (await res.json()).error === 'JWT token expired') {
      session.destroy();
    } else {
      return res;
    }
  });
};

export default internalFetch;
