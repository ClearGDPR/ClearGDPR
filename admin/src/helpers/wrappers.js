import React from 'react';
import PropTypes from 'prop-types';
import lodash from 'lodash';
import { UsersProvider } from 'containers/Users/UsersContext';
import { ProcessorsProvider } from 'containers/Processors/ProcessorsContext';
import { PanelProvider } from 'containers/MainLayout/PanelContext';
import { RectificationsProvider } from 'containers/Rectifications/RectificationsContext';
import { StatsProvider } from 'containers/Dashboard/StatsContext';

export const withWrapper = Wrapper => Component => {
  let wrapper = props => (
    <Wrapper>
      <Component {...props}>{props.children}</Component>
    </Wrapper>
  );
  wrapper.propTypes = {
    children: PropTypes.node
  };
  return wrapper;
};

export const withContextProviders = lodash.flow.apply(null, [
  withWrapper(UsersProvider),
  withWrapper(ProcessorsProvider),
  withWrapper(RectificationsProvider),
  withWrapper(PanelProvider),
  withWrapper(StatsProvider)
]);
