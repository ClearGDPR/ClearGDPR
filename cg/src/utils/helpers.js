const Promise = require('bluebird');
const winston = require('winston');

const { VALID_RUN_MODES } = require('./constants');

function inControllerMode() {
  return process.env.MODE === VALID_RUN_MODES.CONTROLLER;
}

function getMyAddress() {
  return process.env.MY_ADDRESS;
}
function getTokenExpiry() {
  return process.env.JWT_TOKEN_EXPIRY || 604800;
}

const timeout = ms => new Promise(res => setTimeout(res, ms));

async function retryAsync(promise, options = {}) {
  let tries = 0;

  const maxRetries = options.maxRetries || 3;
  const interval = options.interval || 500;
  let delay = 0;

  while (tries < maxRetries) {
    await Promise.delay(delay);
    tries++;
    try {
      return await promise();
    } catch (e) {
      delay = interval * tries;
      winston.info(`Attempt no. ${tries} failed. Next attempt in ${delay}ms`);
      winston.log('debug', `Error: ${e.toString()}`);
    }
  }

  winston.error(`Operation failed after ${tries} attempts`);
}

module.exports = {
  getTokenExpiry,
  inControllerMode,
  getMyAddress,
  timeout,
  retryAsync
};
