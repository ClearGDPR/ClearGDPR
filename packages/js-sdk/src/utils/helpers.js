import { merge } from 'lodash/object';

export function parseJSON(response) {
  const contentType = response.headers.get('Content-Type');

  if (contentType.indexOf('application/json') > -1) {
    return new Promise(resolve =>
      response.json().then(json => {
        return resolve({
          status: response.status,
          ok: response.ok,
          data: json
        });
      })
    );
  } else {
    return new Promise(resolve =>
      response.text().then(text => {
        return resolve({
          status: response.status,
          ok: response.ok,
          data: text
        });
      })
    );
  }
}

export function buildRequestBody(body) {
  return body ? JSON.stringify(body) : null;
}

/**
 * Merges objects values
 * @param {Object} options objects to be merged
 * @returns {Object} New merged object based on params
 */
export function mergeOptions(...objects) {
  return merge(...objects);
}
