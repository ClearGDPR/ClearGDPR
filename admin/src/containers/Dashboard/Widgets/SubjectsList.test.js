import React from 'react';
import SubjectsListContainer from './SubjectsList';
import { shallow, mount } from 'enzyme';
import { flushPromises } from 'tests/helpers/TestUtils';
import session from 'helpers/Session';

jest.mock('helpers/Session');

beforeEach(() => {
  session.getToken.mockReturnValue('token');
});

const subjectsResponse = {
  paging: {
    total: 4,
    current: 1
  },
  data: []
};
describe('(Container) SubjectsList', () => {
  it('Should have the correct state after mounting', async () => {
    global.fetch = jest.fn().mockImplementationOnce(() =>
      Promise.resolve({
        status: 200,
        json: () => Promise.resolve(subjectsResponse)
      })
    );

    const component = mount(<SubjectsListContainer />);
    await flushPromises();

    expect(component.state()).toEqual(
      expect.objectContaining({
        paging: { total: 4, current: 1 },
        subjects: [],
        isLoading: false,
        errorState: false
      })
    );
  });

  it('should render correctly', async () => {
    global.fetch = jest.fn().mockImplementationOnce(() =>
      Promise.resolve({
        status: 200,
        json: () => Promise.resolve(subjectsResponse)
      })
    );

    const component = shallow(<SubjectsListContainer />);

    await flushPromises();
    expect(component).toMatchSnapshot();
  });

  it('Should update state correctly when paging', async () => {
    global.fetch = jest.fn().mockImplementationOnce(() =>
      Promise.resolve({
        status: 200,
        json: () => Promise.resolve(subjectsResponse)
      })
    );

    const component = mount(<SubjectsListContainer />);
    await flushPromises();

    global.fetch = jest.fn().mockImplementationOnce(() =>
      Promise.resolve({
        status: 200,
        json: () =>
          Promise.resolve(Object.assign({}, subjectsResponse, { paging: { total: 4, current: 2 } }))
      })
    );

    component.instance().fetchSubjects(2);
    await flushPromises();

    expect(component.state()).toEqual(
      expect.objectContaining({ paging: { current: 2, total: 4 } })
    );
  });
});
