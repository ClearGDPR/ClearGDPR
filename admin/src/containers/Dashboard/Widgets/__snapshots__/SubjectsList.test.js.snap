// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) SubjectsList should render correctly 1`] = `
<div>
  <SubjectsList
    errorState={false}
    isLoading={false}
    loading={true}
    paging={Object {}}
    subjects={Array []}
  />
  <Paginate
    onPageChange={[Function]}
    pageCount={1}
  />
</div>
`;
